### PR TITLE
Added fix and unit test for queryContentRegexAssertion() method

### PR DESCRIPTION
**Pull request by RalfEggert**
Solves issue for #32.

### CODE REVIEW

File: src/PHPUnit/Controller/AbstractHttpControllerTestCase.php [hunk 1]
```diff
@@ -802,15 +802,22 @@ private function queryContentRegexAssertion($path, $pattern, $useXpath = false)
...
         }
-        if (!preg_match($pattern, $result->current()->nodeValue)) {
+
+        $nodeValues = [];
+
+        foreach ($result as $node) {
+            $nodeValues[] = $node->nodeValue;
+        }
+
+        if (!preg_match($pattern, implode('', $nodeValues))) {
```
Comment (weierophinney) [inline src/PHPUnit/Controller/AbstractHttpControllerTestCase.php]: The concern I have with this approach is that the regex might match across content boundaries.

As an example, consider this markup:

``` html
<div>foo</div>
<div>bar</div>
```

and then the following assertion:

``` php
$this->assertQueryContentRegex('div', '/foobar/');
```

The above, with the change you suggest here, would be a false positive.

I'd suggest doing the regex in the loop itself:

``` php
$found = false;

foreach ($result as $node) {
    if (preg_match($pattern, $node->nodeValue)) {
        $found = true;
        break;
    }
}

if (! $found) {
    throw new PHPUnit_Framework_ExpectationFailedException(/* ... */);
}

$this->assertTrue($found);
```

This approach will resolve the issue you reported in #32, while ensuring we
don't get false positives due to boundary issues.

File: test/PHPUnit/Controller/AbstractHttpControllerTestCaseTest.php [hunk 2]
```diff
@@ -414,6 +414,18 @@ public function testAssertQueryContentRegex()
         $this->assertQueryContentRegex('div#content', '#o{3,}#');
     }
 
+    public function testAssertQueryContentRegexMultipleMatches()
```
Comment (weierophinney) [inline test/PHPUnit/Controller/AbstractHttpControllerTestCaseTest.php:417]: Following on the previous comment, please also add a test that validates that the markup I showed above would not pass this assertion.

**Comment by RalfEggert** [comment]
@weierophinney 

I added another test case to handle unexpected false positives and added for solution. I had to tweak it a little otherwise the exception message had to actual content to output. I hope that tweak is ok.

**Comment by weierophinney** [comment]
Thanks, @RalfEggert! Cherry-picked to master, for release with 3.0.2.

**Pull request closed without merging**